### PR TITLE
fix:code updates init for ERC7739 updates

### PR DESCRIPTION
**Pull request by livingrockrises**

### CODE REVIEW

File: contracts/base/ERC7739Validator.sol [hunk 1]
```diff
@@ -325,4 +355,16 @@ abstract contract ERC7739Validator is IERC7739 {
...
+
+    /// @dev Backwards compatibility stuff
+    /// For automatic detection that the smart account supports the nested EIP-712 workflow.
+    /// By default, it returns `bytes32(bytes4(keccak256("supportsNestedTypedDataSign()")))`,
+    /// denoting support for the default behavior, as implemented in
+    /// `_erc1271IsValidSignatureViaNestedEIP712`, which is called in `isValidSignature`.
+    /// Future extensions should return a different non-zero `result` to denote different behavior.
+    /// This method intentionally returns bytes32 to allow freedom for future extensions.
+    function supportsNestedTypedDataSign() public view virtual returns (bytes32 result) {
+        result = bytes4(0xd620c85a);
```
[the remark this answers is not in the document]
Comment (filmakarov) [reply contracts/base/ERC7739Validator.sol:367]: It was specified on the ERC7739. I believe those are MSB of `keccak256("supportsNestedTypedDataSign")`

File: contracts/base/ERC7739Validator.sol [hunk 2]
```diff
@@ -1,37 +1,56 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.27;
 
-import { IERC7739 } from "../interfaces/IERC7739.sol";
-import { EIP712 } from "solady/utils/EIP712.sol";
-
-/// @title ERC-7739: Nested Typed Data Sign Support for ERC-7579 Validators
+interface IERC5267 {
```
[the remark this answers is not in the document]
Comment (filmakarov) [reply contracts/base/ERC7739Validator.sol:4]: ERC-5267 defines `eip712Domain()` interface

[the remark this answers is not in the document]
Comment (livingrockrises) [reply contracts/base/ERC7739Validator.sol:4]: ok

File: contracts/base/ERC7739Validator.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (livingrockrises) [reply contracts/base/ERC7739Validator.sol:367]: ah. ok. return type in this function is bytes32 result so have to correct that anyways.

wondering why not just boolean but ok I see. check above line

**Review by filmakarov — APPROVED** [review]

**Comment by livingrockrises** [comment]
Screenshot 2024-10-31 at 7 33 08 AM
this did not use to fail.

**Pull request merged**